### PR TITLE
Add solution to Project Euler problem 117

### PR DESCRIPTION
**Pull request by radadiyamohit81**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by radadiyamohit81** [comment]
@cclauss please check my PR

**Comment by radadiyamohit81** [comment]
Hi @Kush1101 , I just contribute to Eular project so I request you to check

File: project_euler/problem_117/sol1.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+How many ways can a row n units long be filled with
+"""
+
+
+def solution(length: int = 50) -> str:
+    """
+    >>> solution(50)
+    '100808458960497'
+    """
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:16]: Please read through the [coding style](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md#coding-style) point 5 about `doctest`.

Comment (radadiyamohit81) [reply project_euler/problem_117/sol1.py:16]: so I just need to add a brief explanation about what does function can do.. AmI right?

**Comment by radadiyamohit81** [comment]
Hi @dhruvmanila I updated my file please check it

**Comment by radadiyamohit81** [comment]
Hi @Kush1101, @dhruvmanila please check my PR .

File: project_euler/problem_117/sol1.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ - Blue tile    (n>=4): Rest of the row can be anything of length n-4. Add ways[n-4].
+"""
+
+
+def solution(length: int = 50) -> str:
+    """
+    This function is used for finding how many units long be filled with
+
+    >>> solution(50)
+    '100808458960497'
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:23]: Please read README.md from `project_euler` directory point 5.

**Comment by radadiyamohit81** [comment]
Hi @Kush1101, @dhruvmanila please check my PR

**Comment by radadiyamohit81** [comment]
Hi @Kush1101, @dhruvmanila please check my PR I updated my file as per requirement.

File: project_euler/problem_117/sol1.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+"""
+Project Euler Problem 104: https://projecteuler.net/problem=117
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:2]:
```suggestion
Project Euler Problem 117: https://projecteuler.net/problem=117
```
This seems like a typo.

File: project_euler/problem_117/sol1.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+Compute n = 0 manually as a base case.
+ - Now assume n >= 1. Look at the leftmost item and sum up the possibilities.
+ - Black square (n>=1): Rest of the row can be anything of length n-1. Add ways[n-1].
+ - Red tile     (n>=2): Rest of the row can be anything of length n-2. Add ways[n-2].
+ - Green tile   (n>=3): Rest of the row can be anything of length n-3. Add ways[n-3].
+ - Blue tile    (n>=4): Rest of the row can be anything of length n-4. Add ways[n-4].
+"""
+
+
+def _calculate(length: int) -> str:
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:18]: This seems unnecessary, please use this function as `solution()` and remove the other function. Also, don't put the test for a solution in `doctest`.

File: project_euler/problem_117/sol1.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    '1490'
+    >>> _calculate(15)
+    '10671'
+    >>> _calculate(41)
+    '274423830033'
+    """
+    ways = [1] + [0] * length
+    for n in range(1, len(ways)):
+        ways[n] += sum(ways[max(n - 4, 0) : n])
+    return str(ways[-1])
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:33]: You don't need to convert it to `str`. That will be done by us when we validate the solution.

File: project_euler/problem_117/sol1.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    >>> solution(25)
+    '7555935'
+    """
+    return _calculate(length)
+
+
+if __name__ == "__main__":
+    from doctest import testmod
+
+    testmod()
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:51]:
```suggestion
```
No need for this.

File: project_euler/problem_117/sol1.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    print(f"{solution()}")
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:52]:
```suggestion
    print(f"{solution() = }")
```

**Review by dhruvmanila — CHANGES_REQUESTED** [review]
Awesome work so far! Just a few small changes and this is good to go.

**Comment by dhruvmanila** [comment]
Please change the title of this pull request according to the submission files.

**Comment by radadiyamohit81** [comment]
Hey @dhruvmanila please check my PR i update all file name, code as per requirements

**Comment by dhruvmanila** [comment]
Please be respectful of names and capitalizations.

File: project_euler/problem_117/sol1.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+ - Green tile   (n>=3): Rest of the row can be anything of length n-3. Add ways[n-3].
+ - Blue tile    (n>=4): Rest of the row can be anything of length n-4. Add ways[n-4].
+"""
+
+
+def solution(length: int = 50) -> str:
+    """
+    This function is used for finding how many units long be filled with
+    >>> solution()
+    100808458960497
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:21]: As mentioned, please read the Solution guidelines always.
Remove this test as mentioned in my previous review and add a test for the function and not for the solution.

File: project_euler/problem_117/sol1.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+Compute n = 0 manually as a base case.
+ - Now assume n >= 1. Look at the leftmost item and sum up the possibilities.
+ - Black square (n>=1): Rest of the row can be anything of length n-1. Add ways[n-1].
+ - Red tile     (n>=2): Rest of the row can be anything of length n-2. Add ways[n-2].
+ - Green tile   (n>=3): Rest of the row can be anything of length n-3. Add ways[n-3].
+ - Blue tile    (n>=4): Rest of the row can be anything of length n-4. Add ways[n-4].
+"""
+
+
+def solution(length: int = 50) -> str:
```
Comment (dhruvmanila) [inline project_euler/problem_117/sol1.py:17]: The return type annotation needs an update.

**Pull request closed without merging**